src/Groups/matrices/MatGrp.jl

```diff
@@ -0,0 +1,892 @@
+# nor whether the two given matrices really match.
+function MatrixGroupElem(G::MatrixGroup, x::MatElem, x_gap::GapObj)
+   z = MatrixGroupElem(G,x)
+   z.X = x_gap
+   return z
+end
+
+
+ring_elem_type(::Type{MatrixGroup{S,T}}) where {S,T} = S
+mat_elem_type(::Type{MatrixGroup{S,T}}) where {S,T} = T
+_gap_filter(::Type{<:MatrixGroup}) = GAP.Globals.IsMatrixGroup
+
+
+
+########################################################################
+#
+# Basic
+#
+########################################################################
+
+function Base.show(io::IO, x::MatrixGroup)
+   AbstractAlgebra.@show_name(io, x)
+   AbstractAlgebra.@show_special(io,x)
+
+   if isdefined(x, :descr)
+      if x.descr==:GU || x.descr==:SU
+         print(io, string(x.descr), "(",x.deg,",",characteristic(x.ring)^(div(degree(x.ring),2)),")")
+      else
+         print(io, string(x.descr), "(",x.deg,",",order(x.ring),")")
+      end
+   else
+      print(io, "Matrix group of degree ", x.deg, " over ")
+      show(IOContext(io, :compact => true), x.ring)
+   end
+end
+
+function Base.show(io::IO, x::MatrixGroupElem)
+   if isdefined(x, :elm)
+      show(io, "text/plain", x.elm)
+#      print(io, x.elm)
+   else
+      print(io, GAP.gap_to_julia(GAP.Globals.StringViewObj(x.X)))
+   end
+end
+
+group_element(G::MatrixGroup, x::GapObj) = MatrixGroupElem(G,x)
+
+function assign_from_description(G::MatrixGroup)
+   if G.descr==:GL G.X=GAP.Globals.GL(G.deg,G.ring_iso.codomain)
+   elseif G.descr==:SL G.X=GAP.Globals.SL(G.deg,G.ring_iso.codomain)
+   elseif G.descr==:Sp G.X=GAP.Globals.Sp(G.deg,G.ring_iso.codomain)
+   elseif G.descr==Symbol("GO+") G.X=GAP.Globals.GO(1,G.deg,G.ring_iso.codomain)
+   elseif G.descr==Symbol("SO+") G.X=GAP.Globals.SO(1,G.deg,G.ring_iso.codomain)
+   elseif G.descr==Symbol("Omega+") G.X=GAP.Globals.Omega(1,G.deg,Int(order(G.ring)))
+   elseif G.descr==Symbol("GO-") G.X=GAP.Globals.GO(-1,G.deg,G.ring_iso.codomain)
+   elseif G.descr==Symbol("SO-") G.X=GAP.Globals.SO(-1,G.deg,G.ring_iso.codomain)
+   elseif G.descr==Symbol("Omega-") G.X=GAP.Globals.Omega(-1,G.deg,Int(order(G.ring)))
+   elseif G.descr==:GO G.X=GAP.Globals.GO(0,G.deg,G.ring_iso.codomain)
+   elseif G.descr==:SO G.X=GAP.Globals.SO(0,G.deg,G.ring_iso.codomain)
+   elseif G.descr==:Omega G.X=GAP.Globals.Omega(0,G.deg,Int(order(G.ring)))
+   elseif G.descr==:GU G.X=GAP.Globals.GU(G.deg,Int(characteristic(G.ring)^(div(degree(G.ring),2) ) ))
+   elseif G.descr==:SU G.X=GAP.Globals.SU(G.deg,Int(characteristic(G.ring)^(div(degree(G.ring),2) ) ))
+   else error("unsupported description")
+   end
+end
+
+# return the G.sym if isdefined(G, :sym); otherwise, the field :sym is computed and set using information from other defined fields
+# NOTE: if G.X has to be set, then also the fields G.ring_iso and G.mat_iso are set
+function Base.getproperty(G::MatrixGroup, sym::Symbol)
+
+   if isdefined(G,sym) return getfield(G,sym) end
```

Comment:
```suggestion
   isdefined(G,sym) && return getfield(G,sym)
```
would be more idiomatic